auth/api/iam/client.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2023 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package iam
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/core"
+)
+
+// HTTPClient holds the server address and other basic settings for the http client
+type HTTPClient struct {
+	httpClient core.HTTPRequestDoer
+}
+
+// NewHTTPClient creates a new api client.
+func NewHTTPClient(config core.ClientConfig) HTTPClient {
+	return HTTPClient{
+		httpClient: core.MustCreateHTTPClient(config, nil),
+	}
+}
+
+func (hb HTTPClient) clientWithBase(baseURL string) ClientInterface {
+	// can only be used for public APIs
+	response, err := NewClientWithResponses(baseURL, WithHTTPClient(hb.httpClient))
+	if err != nil {
+		panic(err)
+	}
```

Comment:
2x panic, is this acceptable(/impossible)? I think the second can be triggered when `baseURL` is not properly validated?

Reply:
An error can only occur from the pass options to `NewClientWithResponses`. We don't pass any that can cause an error. I'll add a comment that when adding options, it should be reassed.